src/plugins/kibana_react/public/empty_state_page/empty_state_page.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { EuiPageTemplate } from '@elastic/eui';
+import React from 'react';
+import { KibanaPageTemplateProps, NO_DATA_PAGE_TEMPLATE_PROPS, NoDataPage } from '../page_template';
+
+interface Props {
+  classes: string;
+}
+
+type EmptyPageProps = Props & KibanaPageTemplateProps;
+
+export const EmptyStatePage = (props: EmptyPageProps) => {
+  const { template, classes, pageSideBar, pageSideBarProps, noDataConfig } = props;
+
+  if (!noDataConfig) {
+    return null;
+  }
+
+  return (
+    <EuiPageTemplate
+      data-test-subj={props['data-test-subj']}
+      template={template}
+      className={classes}
+      pageSideBar={pageSideBar}
+      pageSideBarProps={pageSideBarProps}
+      {...NO_DATA_PAGE_TEMPLATE_PROPS}
```

Comment:
Unfortunately the problem with this refactor is that there is still a lot of logic that alters these template props within KibanaPageTemplate and not just based on `noDataConfig` that haven't been carried over. So if consumers try to use this as a standalone page layout, it won't work properly.

If we know that we don't want consumers to use this as a standalone component, I'd suggest keeping it inside the `page_template` folder. Otherwise, we'll have to look more closely at what else needs to be refactored from the `KibanaPageTemplate` component.

Reply:
As I said, my goal for this iteration was just to split one huge component into smaller units, so that we can get a better sense of what can be pulled/separated. As I mentioned, the goal of this page IS to create `noDataConfig` , rather than take it as a prop. Ideally, the page would take just the solution/app name and then fetch its own render props from some pre-defined registry. Given the props are always the same for one solution, this can be a registry of some sort. But haven't thought through the details of implementation just yet.

Reply:
Ok, so the known technical implication right now will be that consumers still cannot yet use this component as-is, that they will still have to go through `KibanaPageTemplate` until we can do more refactoring?

Reply:
I think with extracting some more logic around class names creation and template name from `PageTemplate` and putting it here, it should be ok to use as a standalone component. Will do that now.

Reply:
I had a look at what would need to be done for this to be a " standalone component". It is basically copying a lot of functionality from `PageTemplate` to this component, and this is not the direction I want to go in. My idea was to create a component that would take (for now) a predefined set of everything that it needs to render, without the component knowing if it's analytics or a solution or something else.  So `sidebarBar` and `sidebarProps` would be defined somewhere outside and passed to the component. Later on, we could create a registry that would hold these props. So the component would receive just the solution/app name and be able to fetch the props it needs to render. I tried elaborating on that a bit more here: https://github.com/elastic/kibana/issues/125664

But for now, I don't see a point of making this a mini `PageTemplate` component.